torchmultimodal/modules/encoders/mdetr_image_encoder.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from collections import OrderedDict
+
+import torch.nn.functional as F
+import torchvision
+from torch import nn
+from torchmultimodal.modules.layers.normalizations import FrozenBatchNorm2d
+from torchmultimodal.utils.common import NestedTensor
+from torchvision.models._utils import IntermediateLayerGetter
+from torchvision.models.resnet import ResNet101_Weights
+
+# At this point this class is basically just a wrapper
+# around the backbone to support tensor list and IntermediateLayerGetter.
+# We should refactor to pass tensor and mask separately
+class MDETRBackbone(nn.Module):
```

Comment:
is the plan to change the name of this class? Since this is a general wrapper around IntermediateLayerGetter and NestedTensor

Reply:
Good point. To make it truly general we can probably pass return_layers as an argument, otherwise this is too specific to MDETR's ResNet implementation.